nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs

```diff
@@ -110,8 +50,73 @@ public void Start()
 
                 _threadWatch.Start();
             }
-        }
-
+        }
+
+        private void StartWatcher()
+        {
+            _ownerManager.OnLogMessageAvailable($"PortSerial device watcher started @ Thread {_threadWatch.ManagedThreadId} [ProcessID: {Process.GetCurrentProcess().Id}]");
+
+            _ports = new List<string>();
+
+            _started = true;

```

Comment:
I know this start pattern with the bool was already existing before. But maybe, while you are improving this, can you create a cancellation token instead?
So you can replace the Thread.Sleep(200) with the cancellation token wait instead. And when stopping, it will not wait for nothing.

Reply:
I think the Polly library was incorporated into the project at a later stage of development. Therefore, many repetitions are still done manually by hand. A lot could still be refactored. I have only changed what caused me an error.
In many places you just have to wait and try again. This is exactly what Polly makes very elegant.

Reply:
For example, the AddDeviceToListAsync method in PortSerialManager.cs does the same thing twice with a random delay in between. Polly could do that too. And the method would be much leaner.
But it wasn't part of that problem.

Reply:
Yes, I noticed how super tricky the timings are. I thought at first this must be a simple bug with the two COM1s, but no, the truth is that finding this bug was very complicated! That's why I tested with many devices at the same time. But I only have ESP32. Maybe you can test it with other devices.

Reply:
Yeah... welcome to the club of the debuggers of the debugger! 🤪 

From my experience: if this works with ESP32 it will work with all the others. ESP32 is the pickiest with timmings. And all other serial connections have better performance. USB wise.